README.md

```diff
@@ -185,6 +177,87 @@ You can navigate to the new HTTPS function's URL printed out by the deploy comma
 For the Realtime Database function, login to the Firebase Console and try
 changing values under `/messages/{randomValue}/original`.
 
+### 7. Scripts (optional)
+
+You can use NPM scripts to simplify the work-flow of serving and deploying functions.
+
+Update your `functions/package.json` to be like so:
+
+```json
+{
+	"...": "...",
+    "scripts": {
+         "build": "pub run build_runner build --output=build",
+         "watch": "pub run build_runner watch --output=build",
+
+        "preserve": "npm run build",
+        "serve": "firebase serve --only functions",
+
+        "predeploy": "npm run build",
+        "deploy": "firebase deploy --only functions",
+
+        "preshell": "npm run build",
+        "shell": "firebase experimental:functions:shell",
+
+        "...": "..."
+    }
+}
+```
+
+## Examples
+
+
+### HTTPS Function
+
+```dart
+import 'package:firebase_functions_interop/firebase_functions_interop.dart';
+
+void main() {
+  functions['helloWorld'] =
+      FirebaseFunctions.https.onRequest(helloWorld);
+}
+
+void helloWorld(ExpressHttpRequest request) {
+  request.response.writeln('Hello world');
+  request.response.close();
+}
+```
+
+### Realtime Database Function
+
+```dart
+void main() {
+  functions['makeUppercase'] = FirebaseFunctions.database
+      .ref('/messages/{messageId}/original')
+      .onWrite(makeUppercase);
+}
+
+FutureOr<void> makeUppercase(DatabaseEvent<String> event) {
+  var original = event.data.val();
+  print('Uppercasing $original');
+  return event.data.ref.parent.child('uppercase').setValue(uppercase);
+}
+```
+
+### Firestore Function
+
+```dart
+void main() {
+  functions['makeNamesUppercase'] = FirebaseFunctions.firestore
+      .document('/users/{userId}').onWrite(makeNamesUppercase)
+}
+
+FutureOr<void> makeNamesUppercase(FirestoreEvent event) {
+  var original = event.data.data.getString("name");
+  print('Uppercasing $original');
+
+  UpdateData newData = new UpdateData();
+  newData.setString("uppercasedName", original);
+
+  return event.data.reference.updateData(newData);
```

Comment:
Wouldn't this produce an infinite loop function uppercasing the same value again and again?
May at least need to check if `uppercasedName` is not present on the document maybe?

Reply:
Good point. I'll fix it in a few minutes!